draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,545 @@ author:
+
+When Proxy-DNS-SVCB (or a cached SVCB) contains an AliasMode record,
+clients SHOULD either use a connection made to the TargetName
+of the AliasMode record, or they SHOULD use a connection
+where any of the hostnames returned in Proxy-DNS-Used
+matches the TargetName of the AliasMode record.
+
+
+## SVCB-required clients
+
+SVCB-required clients must either perform additional SVCB
+DNS requests, or MUST use a proxy that is known and configured
+to always return Proxy-DNS-SVCB and Proxy-DNS-Used headers.
+
+SVCB-required clients SHOULD set high values for "wait" in
+Proxy-DNS-Request and MUST treat the absence of a Proxy-DNS-SVCB
+response header as an error.
+
+SVCB-required clients may also need to make their initial request to
+an authority with no expectation of being able to use that connection.
+Protocols specifying for SVCB-required clients will need to describe
+what clients should use in this case.
+
+
+## Connection coalescing
+
+*TODO*: Do we want to include text on how clients
+can also use Proxy-DNS-Used for connection coalescing?
+(One thing missing there is the full set of IP addresses
+to look for overlaps within an address family.   I'm not
+sure this is worth the complexity.  Or do we remove this
+sub-section entirely?)
+
+
+## Interaction with Alt-Svc
+
+*TODO*: Write this section...
+
+
+## Interaction with SVCB records obtained through the DNS
+
+*TODO*: Write this section...
+
+
+
 # IANA Considerations
 
 ## HTTP Headers {#iana-header}
 
-This document registers the "DNS-SVCB-Keys" and "DNS-SVCB-Params",
+This document registers the "Proxy-DNS-Request" and "Proxy-DNS-SVCB",
 headers in the "Permanent Message Header Field Names"
 <[](https://www.iana.org/assignments/message-headers)>.
 
 ~~~
   +----------------------+----------+--------+---------------+
   | Header Field Name    | Protocol | Status |   Reference   |
   +----------------------+----------+--------+---------------+
-  | DNS-SVCB-Keys        |   http   |  exp   | This document |
+  | Proxy-DNS-Request    |   http   |  exp   | This document |
+  +----------------------+----------+--------+---------------+
+  | Proxy-DNS-SVCB       |   http   |  exp   | This document |
   +----------------------+----------+--------+---------------+
-  | DNS-SVCB-Params      |   http   |  exp   | This document |
+  | Proxy-DNS-Used       |   http   |  exp   | This document |
   +----------------------+----------+--------+---------------+
 ~~~
 
+# Privacy Considerations {#priv-considerations}
+
+Client variation in Proxy-DNS-Request may be a vector for fingerprinting.
+Clients concerned about this may prefer to:
+* Not include "param" and receive all SVCB parameters.
+* Sort the item's parameters by their keys.
+* Limit variation in the values for "wait"
+  and select a wait value of 50, 100, 200,
+  400, 800, or 1600.
+* Not send "version" if implementing only the default
+  (published RFC) version.
+
+*TODO*: Is this a good set of "wait" values?
```

Comment:
Do we need so many, I wonder? We could always just specify one good default to use.

Reply:
Maybe just pick 2-3?  Some will depend on the performance of the proxy to various DNS authorities.  SVCB-required clients may also may want a much higher value here than SVCB-optional.  I'd be happy reducing this set but we may need concrete data to propose a set.